### PR TITLE
[US006, US041] Physic implementation, frontend & backend tests

### PR DESCRIPTION
**Pull request by tomejzen**

### CODE REVIEW

File: GameServer/Models/Vector2.cs [hunk 1]
```diff
@@ -83,7 +83,9 @@ public Vector2 Normalize()
                 throw new Exception("Degenerated vector given. Cannot normalzie zero vector.");
 
             var v = new Vector2(this);
-            v /= Length();
+            if (LengthSquared() != 1)
+                v /= Length();
+
```
Comment (krzysiekprzekwas) [inline GameServer/Models/Vector2.cs]: Any reason for this aside performance one?

File: GameServer/Models/Vector2.cs [hunk 2]
```diff
@@ -131,6 +133,11 @@ public static Vector2 RadianToVector2(double radian)
         {
             return new Vector2(Math.Cos(radian), Math.Sin(radian));
         }
+        
+        public static double Vector2ToRadian(Vector2 vector)
+        {
+            return Math.Asin(vector.Normalize().X);
```
Comment (krzysiekprzekwas) [inline GameServer/Models/Vector2.cs]: It seems like error for me. How can You calculate radian just by X value? What about Y ?

File: GameTests/PhysicTests.cs [hunk 3]
```diff
@@ -130,7 +130,7 @@ public void ShouldCorrectlyCalculateParallelVectorToNormal2()
             var parallelVector = Physic.GetParallelVectorToNormal(vector, normalvector);
 
             // Assert
-            var expectedVector = new Vector2(-0.5f, 0.5f);
+            var expectedVector = new Vector2(-0.5, 0.5);
```
Comment (krzysiekprzekwas) [inline GameTests/PhysicTests.cs]: Great, consistency is important. F**k floats

File: GameServer/Models/Vector2.cs (hunk 1 above)
Comment (tomejzen) [reply GameServer/Models/Vector2.cs]: Only performance.

**Pull request merged**